### PR TITLE
file match has been created to ensure the proper naming of destinatio…

### PR DESCRIPTION
**Pull request by johnathan-rodriguez**
…n files along with allowing regex to be applied through multiple folders and ignores the source folder

### CODE REVIEW

**Comment by johnathan-rodriguez** [comment]
All pytests have been directly based off of the scenarios listed here: https://www.shipyardapp.com/docs/reference/blueprints/blueprint-library/match-type/#regex-match

File: shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py [hunk 1]
```diff
@@ -230,14 +232,11 @@ def find_all_local_file_names(source_folder_name: str = None) -> List[str]:
...
+
     cwd = os.getcwd()
-    if not source_folder_name:
-        search_path = os.path.normpath(f"{os.getcwd()}/**")
-    else:
-        search_path = os.path.normpath(f"{cwd}/{source_folder_name}/**")
-    all_paths = glob.glob(search_path, recursive=True)
-    logger.debug(f"Found {len(all_paths)}.")
-    logger.debug(all_paths)
+    cwd_extension = os.path.normpath(f"{cwd}/{source_folder_name}/**")
```
Comment (wrp801) [inline shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py:237]: Just noting that this works as expected if `''` is provided but not when `None` is provided as the `source_folder_name`

File: shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py [hunk 2]
```diff
@@ -91,14 +89,18 @@ def combine_folder_and_file_name(folder_name: str, file_name: str) -> str:
     Returns:
     str: The combined folder and file name as a single path.
     """
-    return os.path.join(folder_name, file_name)
+
+    combined_name = os.path.normpath(
```
Comment (wrp801) [inline shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py:93]: Why was this changed?

File: shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py [hunk 3]
```diff
@@ -388,3 +390,114 @@ def write_csv_to_file(csv_object: list, file_name: str) -> None:
         writer.writerows(csv_object)
     logger.info(f"CSV data stored at {file_name}")
     return
+
+
+def _duplicate_filename_count(filename, files):
```
Comment (wrp801) [inline shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py:395]: Is this being used?  I didn't see any references so can we delete this?

**Review by wrp801 — COMMENTED** [review]
Just left a few comments that need to be addressed before approving

File: shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py (hunk 2 above)
Comment (johnathan-rodriguez) [reply shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py:93]: join did not handle the leading `/` the same as normpath

Comment (johnathan-rodriguez) [reply shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py:93]: figured more details would probably be helpful
file_name =`/test.txt` 
folder_name = '/parent_folder/'

os.path.join(folder_name,file_name) outputs /test.txt
os.path.normpath(...) outputs parent_folder/test.txt

if the file name didn not have the leading `/` it would come out to the same value
